hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorContainer.java

```diff
@@ -345,58 +320,79 @@ ScheduledFuture createContextAndSchedule(TaskDefinition definition) {
 
     void checkNotDuplicateTask(String taskName) {
         if (tasks.containsKey(taskName)) {
-            throw new DuplicateTaskException("There is already a task "
-                    + "with the same name '" + taskName + "' in '" + getName() + "'");
+            throw new DuplicateTaskException(
+                    "There is already a task " + "with the same name '" + taskName + "' in '" + getName() + "'");
         }
     }
 
     void checkNotAtCapacity() {
         if (capacity != 0 && tasks.size() >= capacity) {
-            throw new RejectedExecutionException("Maximum capacity (" + capacity + ") of tasks reached, "
-                    + "for scheduled executor (" + name + ")");
+            throw new RejectedExecutionException(
+                    "Maximum capacity (" + capacity + ") of tasks reached, " + "for scheduled executor (" + name + ")");
         }
     }
 
     /**
-     * State is published after every run.
-     * When replicas get promoted, they start of, with the latest state see {@link TaskRunner#initOnce()}
+     * State is published after every run. When replicas get promoted, they start of, with the latest state see {@link
+     * TaskRunner#initOnce()}
      */
-    protected void publishTaskState(String taskName, Map stateSnapshot, ScheduledTaskStatisticsImpl statsSnapshot,
-                                    ScheduledTaskResult result) {
+    void publishTaskState(String taskName, Map stateSnapshot, ScheduledTaskStatisticsImpl statsSnapshot,
+                          ScheduledTaskResult result) {
         if (logger.isFinestEnabled()) {
-            logger.finest("[Scheduler: " + name + "][Partition: " + partitionId + "][Task: " + taskName + "] "
-                    + "Publishing state, to replicas. State: " + stateSnapshot);
+            log(FINEST, "Publishing state, to replicas. State: " + stateSnapshot);
         }
 
         Operation op = new SyncStateOperation(getName(), taskName, stateSnapshot, statsSnapshot, result);
-        createInvocationBuilder(op)
-                .invoke()
-                .join();
+        createInvocationBuilder(op).invoke().join();
     }
 
     protected InvocationBuilder createInvocationBuilder(Operation op) {
         OperationService operationService = nodeEngine.getOperationService();
         return operationService.createInvocationBuilder(SERVICE_NAME, op, partitionId);
     }
 
+    protected void log(Level level, String msg) {
+        log(level, null, msg);
+    }
+
+    protected void log(Level level, String taskName, String msg) {
+        log(level, taskName, msg, null);
+    }
+
+    protected void log(Level level, String taskName, String msg, Throwable t) {
+        System.out.println("ENABLED: " + logger.isLoggable(level) + " vs " + logger.isFinestEnabled());
+        if (logger.isLoggable(level)) {
+            StringBuilder log = new StringBuilder();
+            log.append("[Scheduler: " + name + "][Partition: " + partitionId + "]");
```

Comment:
I prefer the one-liner as well, easier to use. 
However, besides styling, the compiler is doing this type of optimization for quite some time now, simple string concatenation is transformed to `SB.append` in the bytecode. StringBuilder makes more sense to dynamic code, like loops etc. Here I used it because of the multiline comments.

Example bytecode output:

```
INVOKESPECIAL java/lang/StringBuilder.<init> ()V
LDC "[Scheduler: "
INVOKEVIRTUAL java/lang/StringBuilder.append (Ljava/lang/String;)Ljava/lang/StringBuilder;
ALOAD 0
GETFIELD com/hazelcast/scheduledexecutor/impl/ScheduledExecutorContainer.name : Ljava/lang/String;
INVOKEVIRTUAL java/lang/StringBuilder.append (Ljava/lang/String;)Ljava/lang/StringBuilder;
LDC "][Partition: "
INVOKEVIRTUAL java/lang/StringBuilder.append (Ljava/lang/String;)Ljava/lang/StringBuilder;
ALOAD 0
GETFIELD com/hazelcast/scheduledexecutor/impl/ScheduledExecutorContainer.partitionId : I
INVOKEVIRTUAL java/lang/StringBuilder.append (I)Ljava/lang/StringBuilder;
LDC "]"
INVOKEVIRTUAL java/lang/StringBuilder.append (Ljava/lang/String;)Ljava/lang/StringBuilder;
INVOKEVIRTUAL java/lang/StringBuilder.toString ()Ljava/lang/String;
INVOKEVIRTUAL java/lang/StringBuilder.append (Ljava/lang/String;)Ljava/lang/StringBuilder;
```